File: Delphi/Delphi/KioskMode.pas
unit KioskMode;

interface

 uses
  System.SysUtils,
  Androidapi.JNI.GraphicsContentViewText,
  FMX.Dialogs,
  Androidapi.Helpers,
  Androidapi.JNI.App,
  Androidapi.JNI.Embarcadero,
  Androidapi.JNI.JavaTypes;

 type

   TKioskMode = class

  private
    activity : JActivity;

  private
    procedure setVisibleMode(enable: boolean);
    procedure setLockTask(start : boolean);

  public
    procedure startKiosMode;
    procedure stopKiosMode;
    constructor Create;


  end;

implementation

{ TKioskMode }

constructor TKioskMode.Create;
begin
    activity := TAndroidHelper.Activity;
end;

procedure TKioskMode.setLockTask(start: boolean);
begin
   if(start) then
     activity.startLockTask()
   else
     activity.stopLockTask();
end;

procedure TKioskMode.setVisibleMode(enable: boolean);
begin
     if Enable then
    begin
       activity.getWindow.getDecorView.setSystemUiVisibility(
            TJView.JavaClass.SYSTEM_UI_FLAG_LAYOUT_STABLE
            or TJView.JavaClass.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
            or TJView.JavaClass.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
            or TJView.JavaClass.SYSTEM_UI_FLAG_HIDE_NAVIGATION
            or TJView.JavaClass.SYSTEM_UI_FLAG_FULLSCREEN or TJView.JavaClass.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
    end else begin
      activity.getWindow.getDecorView.setSystemUiVisibility(TJView.JavaClass.SYSTEM_UI_FLAG_LAYOUT_STABLE
                    or TJView.JavaClass.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                    or TJView.JavaClass.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);
    end;

end;

procedure TKioskMode.startKiosMode;
begin
   
   setLockTask(true);
   setVisibleMode(true);
end;

procedure TKioskMode.stopKiosMode;
begin
   setLockTask(false);
   setVisibleMode(false);
end;

end.

File: Android/KioskMode/src/main/java/br/com/estudos/kioskmode/KioskMode.java
/*======================================================================================
  Arquivo       : KioskMode.java
  Projeto       : Modo Quiosque
  Plataforma    : Android
  Equipamentos  : Dispositivos Android
  Data Criação  : 03/Nov/2020
  Autor         : Geovani Nogueira Santos

  Descrição   : KioskMode.java:
                   - Habilita e Desabilita o Kiosk Mode via aplicação.
  =========================================================================================*/

package br.com.estudos.kioskmode;

import android.app.Activity;
import android.app.admin.DevicePolicyManager;
import android.content.ComponentName;

import android.view.View;

public class KioskMode extends AdminReceiver {

    private ComponentName componentName;
    private DevicePolicyManager devicePolicyManager;
    private Activity activity;

    public KioskMode(Activity activity) {
        this.activity = activity;
    }

    public void startKiosMode(){
        this.componentName = new ComponentName(this.activity.getApplicationContext(), AdminReceiver.class);
        this.devicePolicyManager = (DevicePolicyManager) this.activity.getSystemService(this.activity.DEVICE_POLICY_SERVICE);
        this.setLockTask(true);
        this.setVisibleMode(true);
    }

    public void stopKioskMode() throws Exception {
        if(this.componentName == null){
            throw new Exception("Modo Quiosque ainda não foi ativado");
        }
        this.setLockTask(false);
        this.setVisibleMode(false);
        this.componentName = null;
        this.devicePolicyManager = null;
    }

    // Modo que a aplicação ficará visivel no e equipamento
    private void setVisibleMode(Boolean enable) {
        if (enable) {
            int flags = (View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                    | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                    | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
                    | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                    | View.SYSTEM_UI_FLAG_FULLSCREEN
                    | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
            this.activity.getWindow().getDecorView().setSystemUiVisibility(flags);
        } else {
            int flags = (View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                    | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                    | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);
            this.activity.getWindow().getDecorView().setSystemUiVisibility(flags);
        }
    }

    private void setLockTask(boolean start){
        if(start){
            activity.startLockTask();
        }else{
            activity.stopLockTask();
        }
    }

}
